contracts/src/v0.8/ccip/offRamp/MessageTransformerOffRamp.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {IMessageTransformer} from "../interfaces/IMessageTransformer.sol";
+import {Internal} from "../libraries/Internal.sol";
+import {OffRamp} from "./OffRamp.sol";
+
+/// @notice OffRamp that uses a message transformer to transform messages before execution
+contract MessageTransformerOffRamp is OffRamp {
+  address internal s_messageTransformer;
+
+  constructor(
+    StaticConfig memory staticConfig,
+    DynamicConfig memory dynamicConfig,
+    SourceChainConfigArgs[] memory sourceChainConfigs,
+    address messageTransformerAddr
+  ) OffRamp(staticConfig, dynamicConfig, sourceChainConfigs) {
+    if (address(messageTransformerAddr) == address(0)) {
+      revert ZeroAddressNotAllowed();
+    }
+    s_messageTransformer = messageTransformerAddr;
+  }
+
+  function getMessageTransformerAddress() external view returns (address) {
+    return s_messageTransformer;
```

Comment:
missing setMessageTransformer

Reply:
Is it needed? 

When a new ramp needs to be introduced, do we usually deploy a new ramp and associate it with existing and other dependencies? If so, I believe we can follow the same convention when a new message transformer is needed and just redeploy a new ramp with new message transformer.